src/lib/redux-cozy-api.js

```diff
@@ -0,0 +1,484 @@
+    case RECEIVE_UPDATE_CONFIRM:
+      return {
+        ...state,
+        [action.entity.type]: Object.assign({}, state[action.entity.type], objectifyEntitiesArray(action.response.data))
+      }
+    case RECEIVE_DELETION_CONFIRM:
+      return {
+        ...state,
+        [action.entity.type]: removeObjectProperty(state[action.entity.type], action.entity.id)
+      }
+    default:
+      return state
+  }
+}
+
+// endpoint reducers
+const type = (state = null, action) => {
+  switch (action.type) {
+    case FETCH_DOCUMENTS:
+      return action.doctype
+    case FETCH_REFERENCED_FILES:
+      return FILES_DOCTYPE
+    default:
+      return state
+  }
+}
+
+const fetchStatus = (state = 'pending', action) => {
+  switch (action.type) {
+    case FETCH_DOCUMENTS:
+    case FETCH_REFERENCED_FILES:
+      return action.skip > 0 ? 'loadingMore' : 'loading'
+    case RECEIVE_DATA:
+    case RECEIVE_REFERENCED_FILES:
+      return 'loaded'
+    case RECEIVE_ERROR:
+      return 'failed'
+    default:
+      return state
+  }
+}
+
+const lastFetch = (state = null, action) => {
+  switch (action.type) {
+    case RECEIVE_DATA:
+    case RECEIVE_REFERENCED_FILES:
+      return Date.now()
+    default:
+      return state
+  }
+}
+
+const hasMore = (state = false, action) => {
+  switch (action.type) {
+    case RECEIVE_DATA:
+    case RECEIVE_REFERENCED_FILES:
+      const response = action.response
+      return response.next !== undefined ? response.next : state
+    default:
+      return state
+  }
+}
+
+const count = (state = 0, action) => {
+  switch (action.type) {
+    case RECEIVE_DATA:
+    case RECEIVE_REFERENCED_FILES:
+      const response = action.response
+      return response.meta && response.meta.count ? response.meta.count : response.data.length
+    case ADD_REFERENCED_FILES:
+      return state + action.ids.length
+    case REMOVE_REFERENCED_FILES:
+      return state - action.ids.length
+    default:
+      return state
+  }
+}
+
+const ids = (state = [], action) => {
+  switch (action.type) {
+    case RECEIVE_DATA:
+    case RECEIVE_REFERENCED_FILES:
+      if (!action.skip) {
+        return action.response.data.map(doc => doc.id)
+      }
+      return [
+        ...state,
+        ...action.response.data.map(doc => doc.id)
+      ]
+    case ADD_REFERENCED_FILES:
+      // TODO: here we don't follow JSONAPI conventions... (we pass an array of ids in the action)
+      return [
+        ...state,
+        ...action.ids
+      ]
+    case REMOVE_REFERENCED_FILES:
+      return state.filter(id => action.ids.indexOf(id) === -1)
+    case RECEIVE_CREATION_CONFIRM:
+      return [
+        ...state,
+        action.response.data[0].id
+      ]
+    case RECEIVE_DELETION_CONFIRM:
+      return state.filter(id => id !== action.response.data[0].id)
+    default:
+      return state
+  }
+}
+
+const endpoint = combineReducers({
+  type,
+  fetchStatus,
+  lastFetch,
+  hasMore,
+  count,
+  ids
+})
```

Comment:
The endpoint reducers and the associated actions do not take a query parameter. So it will "only" work to fetch all documents in a collection. Is this case so uncommon that we do not want it in this lib (I am under the impression that this lib will be used by other cozy projects at some point) ?